File: usr/src/usr.bin/f77/libF77/c_sin.c
/*-
 * Copyright (c) 1980 The Regents of the University of California.
 * All rights reserved.
 *
 * %sccs.include.proprietary.c%
 */

#ifndef lint
static char sccsid[] = "@(#)c_sin.c	5.2 (Berkeley) %G%";
#endif /* not lint */

#include "complex"

c_sin(r, z)
complex *r, *z;
{
double sin(), cos(), sinh(), cosh();

r->real = sin(z->real) * cosh(z->imag);
r->imag = cos(z->real) * sinh(z->imag);
}

File: .ref-Research-V6/usr/source/fort/rt1/csin.f
	function csin(z1)
	complex csin, z1, z2
	dimension dummy(2)
	equivalence (a,z2,dummy(1)), (b,dummy(2))

	z2 = z1
	c = exp(b)
	d = 1.0/c
	b = cos(a)*(c-d)/2.0
	a = sin(a)*(c+d)/2.0
	csin = z2
	return
	end
